src/Halogen/Aff/Driver.purs

```diff
@@ -43,12 +44,17 @@ import Halogen.Query.HalogenM (HalogenM(..), HalogenF(..), HalogenAp(..))
 
 import Halogen.Aff.Driver.State (ComponentType(..)) as Exports
 
--- | Type alias for driver functions generated by `runUI` - a driver takes an
--- | input of the query algebra (`f`) and returns an `Aff` that returns when
--- | query has been fulfilled.
-type Driver f eff = f ~> Aff (HalogenEffects eff)
-
-type DSL s f m eff p o = HalogenF s f m p o (Aff (HalogenEffects eff))
+-- | A record produced when the root component in a Halogen UI has been run.
+-- | `query` allows external sources to query the root component and `subscribe`
+-- | allows external consumers to receive messages raised by the root component.
+type HalogenIO f o m =
```

Comment:
No, the driver stuff isn't re-exported from `Halogen`, and `HalogenIO`'s types are not driver spectific either.